src/test/java/co/rsk/federate/watcher/FederationWatcherListenerImplTest.java

```diff
@@ -87,7 +89,34 @@ void triggerClientChange_whenExceptionOccurs_shouldHandleException() {
         doThrow(new RuntimeException("Simulated exception")).when(btcToRskClientActive).stop();
 
         // Act & Assert
-        assertDoesNotThrow(() -> federationWatcherListener.onActiveFederationChange(FIRST_FEDERATION));
+        assertDoesNotThrow(() -> federationWatcherListener.onActiveFederationChange(FEDERATION));
+    }
+
+    @Test
+    void onProposedFederationChange_whenNewProposedFederationIsNull_shouldNotStartClient() {
+        // Act
+        federationWatcherListener.onProposedFederationChange(null);
+
+        // Assert
+        verify(btcReleaseClient, never()).start(any(Federation.class));
+    }
+
+    @Test
+    void onProposedFederationChange_whenNewProposedFederationIsValid_shouldStartClient() {
+        // Act
+        federationWatcherListener.onProposedFederationChange(FEDERATION);
+
+        // Assert
+        verify(btcReleaseClient).start(FEDERATION);
+    }
```

Comment:
I think we are missing some test cases, I'd list some that comes to my mind:

- "When the federation member does not belong to the proposed federation..."
- "When btcReleaseClient has already been started by the active federation"
- "When there is a retiring federation and active fed"

Wdyt?

Reply:
I think those tests make more sense to put it on the [FederationWatcherTest](https://github.com/rsksmart/powpeg-node/blob/feat/powpeg_validation_protocol-phase4/src/test/java/co/rsk/federate/watcher/FederationWatcherTest.java) class. This just tests what the impl of `onProposedFederationChange` does.

Reply:
I think it does not hurt to test it here as well. Let's see what the team thinks

Reply:
I have included them on this PR, please check them out